action/src/run.ts

```diff
@@ -24,13 +24,31 @@ import {
 import { buildComparadiseUrl } from './build-comparadise-url';
 import { disableAutoMerge } from './disableAutoMerge';
 
+const VISUAL_TEST_EXECUTION_FAILURE =
+  'Visual tests failed to execute successfully. Perhaps one failed to take a screenshot?';
+const VISUAL_TESTS_PASSED = 'All visual tests passed, and no diffs found!';
+
 export const run = async () => {
   const runAttempt = Number(process.env.GITHUB_RUN_ATTEMPT);
   const isRetry = runAttempt > 1;
   const visualTestCommands = getMultilineInput('visual-test-command', {
     required: true
   });
-  const commitHash = getInput('commit-hash', { required: true });
+  const commitHash = getInput('commit-hash');
+  const diffId = getInput('diff-id');
+
+  if (!commitHash && !diffId) {
+    setFailed('Please provide either a commit-hash or a diff-id.');
+    return;
+  }
+
+  if (commitHash && diffId) {
+    setFailed(
+      'You cannot provide both commit-hash and diff-id. Please choose one.'
+    );
+    return;
+  }
+
```

Comment:
I would prefer to make a generic hash variable like
```ts
const hash = diffId || commitHash;
```
and use this as the hash so we don't have to duplicate the code in this file

Reply:
And then skip all GH calls if `commitHash` is not provided